<launch>

  <arg name="GUI" default="true" />
  <arg name="PROCESS" default="true" />

  <rosparam subst_value="true">
    /camera/rgb_camera/brightness: 20
    /camera/rgb_camera/contrast: 60
    /camera/rgb_camera/enable_auto_white_balance: false
    /camera/rgb_camera/white_balance: 2800
  </rosparam>

  <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" if="$(arg PROCESS)">
  </include>

  <group ns="/camera/color">
    <node name="image_rect_color_passthrough"
          pkg="jsk_topic_tools" type="passthrough"
          clear_params="true">
      <remap from="~input" to="image_raw" />
      <rosparam subst_value="true">
        default_duration: 0
      </rosparam>
    </node>
    <node name="image_view"
          pkg="image_view" type="image_view"
          clear_params="true" if="$(arg GUI)">
      <remap from="image" to="image_rect_color" />
    </node>
  </group>
  <group ns="/camera/aligned_depth_to_color">
    <node name="depth_to_rgb"
          pkg="reorientbot_ros" type="depth_to_rgb.py"
          clear_params="true">
      <remap from="~input/camera_info" to="camera_info" />
      <remap from="~input/depth" to="image_raw" />
    </node>
    <group ns="depth_to_rgb" if="$(arg GUI)">
      <node name="image_view"
            pkg="image_view" type="image_view"
            clear_params="true">
        <remap from="image" to="output/jet" />
      </node>
    </group>
  </group>

  <!--
  <node name="marker_6dof"
        pkg="jsk_interactive_marker" type="marker_6dof"
        clear_params="true" if="$(arg PROCESS)">
    <rosparam subst_value="true">
      frame_id: panda_link8  # parent
      tf_frame: camera_link  # child
      object_type: cube
      publish_tf: true
      tf_duration: 0.01  # 100Hz
      object_x: 0.1
      object_y: 0.1
      object_z: 0.1
      initial_x: 0.056
      initial_y: 0.025
      initial_z: 0.036
      initial_orientation: [0.243, -0.666, 0.281, 0.647]
    </rosparam>
  </node>
  -->
  <include file="$(find reorientbot_ros)/launch/eye_in_hand_camera_pose.launch">
  </include>

</launch>
